apis/workloads/v1alpha1/replicatedstatemachine_types.go

```diff
@@ -23,26 +23,84 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
 
-const (
-	ReplicatedStateMachineKind = "ReplicatedStateMachine"
-)
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// RsmTransformPolicy defines rsm transform type
-// ToSts and ToPod is supported
-// +enum
-// +kubebuilder:validation:Enum={ToPod,ToSts}
-type RsmTransformPolicy string
+type InstanceTemplate struct {
+	// Number of replicas of this template.
+	// Default is 1.
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
 
-const (
-	ToSts RsmTransformPolicy = "ToSts"
-	ToPod RsmTransformPolicy = "ToPod"
-)
+	// Defines the name of the instance.
+	// Only applied when Replicas is 1.
+	// +optional
+	Name *string `json:"name,omitempty"`
```

Comment:
why not using `instanceName`, `Name` gave me the illusion of being a template name

Reply:
This name should be unique, right？